http/cookies.lua

```diff
@@ -0,0 +1,356 @@
+local http_patts = require "lpeg_patterns.http"
+local util = require "http.util"
+local lpeg = require "lpeg"
+local psl = require "psl"
+
+local Set_Cookie_anchored = http_patts.Set_Cookie * lpeg.P(-1)
+
+local function parse_set_cookie(text_cookie, host, path, time)
+	assert(time, "missing time value for cookie parsing")
+	local key, value, matched_cookie = Set_Cookie_anchored:match(text_cookie)
+	if not key then
+		return nil, "cookie did not properly parse"
+	end
+	local cookie = {
+		creation = time;
+		last_access = time;
+		persistent = not not (matched_cookie.expires or matched_cookie["max-age"]);
+		domain = matched_cookie.domain or host;
+		path = matched_cookie.path or path;
+		secure = matched_cookie.secure or false;
+		http_only = matched_cookie.httponly or false;
+		key = key;
+		value = value;
+		host_only = not not matched_cookie.domain;
+		same_site = matched_cookie.same_site;
+	}
+	local age = matched_cookie["max-age"]
+	if age then
+		local is_negative, match = age:match("^(-?)(%d+)$")
+		if is_negative then
+			-- RFC 6265 section 5.2.2 - if the value when converted to an
+			-- integer is negative, the expiration should be the earliest
+			-- representable expiration time.
+			cookie.expires = 0
+		elseif not match then
+			return nil, "expected [-]DIGIT* for max-age field"
+		else
+			cookie.expires = time + tonumber(match, 10)
+		end
+	else -- luacheck: ignore
+		-- ::TODO:: make use of `expires` cookie value
```

Comment:
Plans to implement this TODO?

Reply:
Yes, after I've got all the other bugs done and I think of a decent way to do it. I plan on getting rid of every 'TODO' before it's merged.

Reply:
Sorry, didn't see "plans", saw "plan". I'm thinking just use an RFC pattern to match the values and then use `os.time()` on the values, but I need to look on what common implementations support for patterns.

Reply:
(as a note, while it's on my mind, there's *several* various patterns that could match the "time", which makes it difficult to match - it should be mentioned somewhere)